python/ray/ml/train/trainer.py

```diff
@@ -0,0 +1,77 @@
+import abc
+import logging
+from typing import Dict, Union, Callable, Optional, TYPE_CHECKING
+
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.ml.result import Result
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.tune.trainable import ConvertibleToTrainable
+from ray.util import PublicAPI
+
+if TYPE_CHECKING:
+    from ray.data import Dataset
+
+GenDataset = Union["Dataset", Callable[[], "Dataset"]]
+
+
+logger = logging.getLogger(__name__)
+
+
+@PublicAPI(stability="alpha")
+class Trainer(ConvertibleToTrainable, abc.ABC):
+    """Defines interface for distributed training on Ray.
+    Args:
+        scaling_config: Configuration for how to
```

Comment:
nit: please fix line wrapping here and below.

Reply:
Fixed